components/script/dom/headers.rs

```diff
@@ -220,18 +223,37 @@ impl Headers {
     }
 }
 
-// TODO
-// "Content-Type" once parsed, the value should be
-// `application/x-www-form-urlencoded`, `multipart/form-data`,
-// or `text/plain`.
-// "DPR", "Downlink", "Save-Data", "Viewport-Width", "Width":
-// once parsed, the value should not be failure.
+fn is_cors_safelisted_request_content_type(value: &[u8]) -> Result<bool, Error> {
+    let value_string = str::from_utf8(value);
+    if value_string.is_err() {
+        return Err(Error::Type("Value cannot be transformed into a string.".to_string()))
+    } else {
+        let value_without_parameter = value_string.unwrap().split(";").collect::<Vec<&str>>()[0];
+        match value_without_parameter {
+            "application/x-www-form-urlencoded" |
+            "multipart/form-data" |
+            "text/plain" => Ok(true),
```

Comment:
Yes. We should parse the value first, and operate on the parsed `Header` instance, matching on its type.